syntax = "proto2";

package e2e_noa;
  

// Not using planning.namespace.

enum LaneChangeStage {
  LCS_NONE = 0;
  LCS_WAITING = 1;
  LCS_EXECUTING = 2;
  LCS_PAUSE = 3;
  LCS_RETURN = 4;
}

enum PushState {
  NONE_PUSH = 0;
  LEFT_PUSH = 1;
  RIGHT_PUSH = 2;
  CONGESTION_LEFT_PUSH = 3;
  CONGESTION_RIGHT_PUSH = 4;
}

message LaneChangeStateProto {
  optional LaneChangeStage stage = 1;
  optional bool crossed_boundary = 9;
  optional bool entered_target_lane = 3;
  // Whether this is a lane change to left side.
  optional bool lc_left = 5;
  optional bool force_merge = 8 [default = false];
  optional PushState push_state = 10;
  optional PushState pre_push_state = 11;

  reserved 2, 4, 6, 7;

  // Next ID: 9
}